connection.go

```diff
@@ -107,6 +108,10 @@ func New(addr string, spec *iso8583.MessageSpec, mlReader MessageLengthReader, m
 		}
 	}
 
+	if err := opts.Validate(); err != nil {
+		return nil, fmt.Errorf("validating options: %w", err)
+	}
```

Comment:
This doesn't seem related to adding `*Ctx()` methods. What is the need here? Will it break existing behavior?

Reply:
the idea here is to verify that either the Ctx methods or the original methods are configured but not both. But with rewriting the methods to only have one implementation should take care of this, so I'll remove it.